src/main/java/org/veupathdb/service/eda/ds/plugin/standalonemap/markers/OverlaySpecification.java

```diff
@@ -0,0 +1,59 @@
+package org.veupathdb.service.eda.ds.plugin.standalonemap.markers;
+
+import org.veupathdb.service.eda.generated.model.APIVariableType;
+import org.veupathdb.service.eda.generated.model.CategoricalOverlayConfig;
+import org.veupathdb.service.eda.generated.model.ContinousOverlayConfig;
+import org.veupathdb.service.eda.generated.model.OverlayConfig;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.veupathdb.service.eda.generated.model.OverlayType.CATEGORICAL;
+
+public class OverlaySpecification {
+  private final List<String> labels;
+  private final List<NormalizedBinRange> binRanges;
+  private OverlayRecoder overlayRecoder;
+
+  public OverlaySpecification(OverlayConfig overlayConfig, Function<VariableSpec, String> varTypeFinder) {
+    final String variableType = varTypeFinder.apply(overlayConfig.getOverlayVariable());
```

Comment:
is there anything stopping someone from trying something like claiming a continuous overlay type and passing bin start and ends and labels like 'blue' ? i mean i dont think well actually see that, but .. idk thought it might be worth mentioning

Reply:
Would it be better if we just inferred the labels as "${start}-${end}" for continuous? I'm not sure whether the client will need to specify labels for bin ranges

Reply:
Good question. Right now the R fxn that makes the bins will provide exact values for start and end, but more friendly/ rounded values in the labels.